mmcv/ops/csrc/pytorch/mlu/focal_loss_sigmoid_mlu.cpp

```diff
@@ -0,0 +1,165 @@
+              "But now is ", target.dim(), ".");
+
+  TORCH_CHECK(input.size(0) == target.size(0), "Element num of target should be ", 
+              input.size(0), ". But now is ", target.size(0), ".");
+
+  TORCH_CHECK(output.dim() == 2, "Dimension num of target should be 2. ",
+              "But now is ", output.dim(), ".");
+
+  TORCH_CHECK(input.size(0) == output.size(0) && input.size(1) == output.size(1),
+              "Shape of output and input must be euqal, but now output is ", 
+              output.size(0), ", ", output.size(1), " and input is ",
+              input.size(0), ", ", input.size(1), ".");
+
+  // check dtype
+  TORCH_CHECK(input.scalar_type() == at::kFloat || input.scalar_type() == at::kHalf, 
+              "Data type of input should be Float or Half. But now input type is ",
+              input.scalar_type(), ".");
+
+  TORCH_CHECK(target.scalar_type() == at::kLong, "target type should be Long. ",
+              "But now target type is ", target.scalar_type(), ".");
+
+  TORCH_CHECK(output.scalar_type() == input.scalar_type(), 
+              "Data types of input and output should be the same. But now input type is ",
+              input.scalar_type(), ", output type is ", output.scalar_type(), ".");
+
+  // check weight
+  if (weight.data_ptr() != nullptr) {
+    TORCH_CHECK(weight.scalar_type() == input.scalar_type(), 
+                "Data types of input and weight should be the same. But now input type is ",
+                input.scalar_type(), ", weight type is ", weight.scalar_type(), ".");
+
+    TORCH_CHECK(weight.dim() == 1, "Dimension num of weight should be 1.",
+                "But now is ", weight.dim(), ".");
+
+    TORCH_CHECK(weight.size(0) == input.size(1), "Element num of weight should be ", 
+                input.size(1), ". But now is ", weight.size(0), ".");
+  } else {
+    CNLOG(INFO) << "weight is a empty tensor.";
+  }
+}
+
+void SigmoidFocalLossForwardMLUKernelLauncher(Tensor input,
+                                              Tensor target,
+                                              Tensor weight,
+                                              Tensor output,
+                                              float gamma,
+                                              float alpha) {
+
+  // params check
+  TORCH_CHECK(gamma >= 0, "gamma should be greater than or equal to 0. ",
+                          "But now gamma is ", gamma, ".");
+
+  checkFocalLossSigmoidForwardValidation(input, target, weight, output);
+
+  // check C
+  auto input_N = input.size(0);
+  auto input_C = input.size(1);
+  auto split_target_num = 2;
+  int split_pipeline_num = 6;
+  split_pipeline_num = weight.data_ptr() == nullptr? split_pipeline_num : split_pipeline_num + 1;
+  auto nram_size = torch_mlu::getDeviceAttr(cnrtAttrNramSizePerMcore);
+
+  // target supports only INT on MLU device
+  // while it keeps LONG on host side, so target.itemsize()/2
+  auto threshold_C = (nram_size - NFU_ALIGN_SIZE -
```

Comment:
What is `\ NFU_ALIGN_SIZE * NFU_ALIGN_SIZE` for?

Reply:
I think it is to round down to an interger, which is a multiples of `NFU_ALIGN_SIZE`,for the requirement of alignment. @grimoire

Reply:
This is equal to pad_down function: (x/y)*y